library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleQueue.java

```diff
@@ -0,0 +1,159 @@
+    this.overridingDrmInitData = overridingDrmInitData;
+  }
+
+  void setCurrentLoadingChunk(HlsMediaChunk chunk) {
+    double tolerance = (chunk.endTimeUs - chunk.startTimeUs) * MAX_TIMESTAMP_DEVIATION_PERCENTAGE;
+    this.lowestTimeUs = (long) (chunk.startTimeUs - tolerance);
+    this.highestTimeUs = (long) (chunk.endTimeUs + tolerance);
+    this.chunk = chunk;
+    lastValidTimeUs = C.TIME_UNSET;
+  }
+
+  public void setDrmInitData(@Nullable DrmInitData drmInitData) {
+    this.drmInitData = drmInitData;
+    invalidateUpstreamFormatAdjustment();
+  }
+
+  @SuppressWarnings("ReferenceEquality")
+  @Override
+  public Format getAdjustedUpstreamFormat(Format format) {
+    @Nullable
+    DrmInitData drmInitData = this.drmInitData != null ? this.drmInitData : format.drmInitData;
+    if (drmInitData != null) {
+      @Nullable
+      DrmInitData overridingDrmInitData = this.overridingDrmInitData.get(drmInitData.schemeType);
+      if (overridingDrmInitData != null) {
+        drmInitData = overridingDrmInitData;
+      }
+    }
+    @Nullable Metadata metadata = getAdjustedMetadata(format.metadata);
+    if (drmInitData != format.drmInitData || metadata != format.metadata) {
+      format = format.buildUpon().setDrmInitData(drmInitData).setMetadata(metadata).build();
+    }
+    return super.getAdjustedUpstreamFormat(format);
+  }
+
+  /**
+   * Strips the private timestamp frame from metadata, if present. See:
+   * https://github.com/google/ExoPlayer/issues/5063
+   */
+  @Nullable
+  private Metadata getAdjustedMetadata(@Nullable Metadata metadata) {
+    if (metadata == null) {
+      return null;
+    }
+    int length = metadata.length();
+    int transportStreamTimestampMetadataIndex = C.INDEX_UNSET;
+    for (int i = 0; i < length; i++) {
+      Metadata.Entry metadataEntry = metadata.get(i);
+      if (metadataEntry instanceof PrivFrame) {
+        PrivFrame privFrame = (PrivFrame) metadataEntry;
+        if (HlsMediaChunk.PRIV_TIMESTAMP_FRAME_OWNER.equals(privFrame.owner)) {
+          transportStreamTimestampMetadataIndex = i;
+          break;
+        }
+      }
+    }
+    if (transportStreamTimestampMetadataIndex == C.INDEX_UNSET) {
+      return metadata;
+    }
+    if (length == 1) {
+      return null;
+    }
+    Metadata.Entry[] newMetadataEntries = new Metadata.Entry[length - 1];
+    for (int i = 0; i < length; i++) {
+      if (i != transportStreamTimestampMetadataIndex) {
+        int newIndex = i < transportStreamTimestampMetadataIndex ? i : i - 1;
+        newMetadataEntries[newIndex] = metadata.get(i);
+      }
+    }
+    return new Metadata(newMetadataEntries);
+  }
+
+  @Override
+  public void sampleMetadata(long timeUs, int flags, int size, int offset,  @Nullable CryptoData cryptoData) {
+    // TODO - chunkless prepare, sampleQueue list is not yet initialized for first chunk
+//    Assertions.checkNotNull(chunk, "sampleMetadata without a MediaChunk?");
+    if (chunk == null) {
```

Comment:
Had to replace this with null check, the initial segment loads are started before `HlsSampleStreamWrapper#createSampleQueue()` is called so need a way to get the loading chunk in that method and set it in the `HlsSampleQueue`

Or we ignore checking the first segment loads.